nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -365,22 +326,45 @@ in
     virtualisation.bootDevice =
       mkOption {
         type = types.path;
+        default = lookupDriveDeviceName "root" cfg.qemu.drives;
+        defaultText = literalExpression ''lookupDriveDeviceName "root" cfg.qemu.drives'';
         example = "/dev/vda";
         description =
           lib.mdDoc ''
-            The disk to be used for the root filesystem.
+            The disk to be used for the boot filesystem.
+            By default, it is the same disk as the root filesystem.
+          '';
+        };
```

Comment:
Given this changes the meaning of the option (which it was sorta poorly named to begin with), I feel like we should at a minimum include release notes describing the change and addition of `virtualisation.rootDevice`

I also wonder if we should still rename the option anyway so we can add `mkRemovedOption` and give a useful error for any downstream users of it (I don't imagine there are many downstream users though, if any at all, so I do not feel strnogly on it)

Reply:
We could rename the option but do you have a suggestion on the rename?

Reply:
Maybe `bootLoaderDevice` or `bootDiskDevice`? Idk really, and I wish the `bootDevice` option hadn't already existed before this PR and meant "root device" (but oh well I guess)

Reply:
`bootLoaderDevice` sounds good because it's really a "bootloader device" and lead to probably do a refactor `useBootLoader` -> `bootloader.{enable,device,enableEFI}`, etc. Thank you!

Reply:
Applied!